src/app/sessions/book/page.tsx

```diff
@@ -53,8 +56,19 @@ export default function BookSessionPage() {
       if (response.ok) {
         const { id } = await response.json();
         router.push(`/booking-confirmation/${id}`);
-      } else {
-        throw new Error("Failed to confirm booking");
+      } else if (response.status === 409) {
+        const responseData = await response.json();
+        if (responseData.message === "Game session at max capacity") {
+          for (let i = 0; i < cart.length; i++) {
+            const session = cart[i];
+            if (!data!.some((dataSession) => dataSession.id === session.id)) {
+              handleSessionLimitReached(session.id);
+            }
+            console.log("a Session is full");
+          }
+        } else {
+          throw new Error("Failed to confirm booking");
+        }
       }
```

Comment:
need to redirect to /sessions on error

not sure if this is a better approach but you could also change the sessions on the sessions page with an useeffect hook

Reply:
i think its not working because you're not revalidating the sessions.